compiler/scalc/print.ml

```diff
@@ -165,20 +169,35 @@ and format_block
     (format_statement decl_ctx ~debug)
     fmt block
 
-let format_scope
-    (decl_ctx : decl_ctx)
-    ?(debug : bool = false)
-    (fmt : Format.formatter)
-    (body : scope_body) : unit =
-  if debug then () else ();
-  Format.fprintf fmt "@[<hov 2>%a@ %a@ %a@ %a@]@\n@[<v 2>  %a@]" Print.keyword
-    "let" TopLevelName.format_t body.scope_body_var
-    (Format.pp_print_list
-       ~pp_sep:(fun fmt () -> Format.fprintf fmt "@ ")
-       (fun fmt ((name, _), typ) ->
-         Format.fprintf fmt "%a%a %a@ %a%a" Print.punctuation "("
-           format_local_name name Print.punctuation ":" (Print.typ decl_ctx) typ
-           Print.punctuation ")"))
-    body.scope_body_func.func_params Print.punctuation "="
-    (format_block decl_ctx ~debug)
-    body.scope_body_func.func_body
+let format_item decl_ctx ?debug ppf def =
+  Format.pp_open_hvbox ppf 2;
+  Format.pp_open_hovbox ppf 4;
+  Print.keyword ppf "let ";
+  let () =
+    match def with
+    | SVar { var; expr } ->
+      format_var_name ppf var;
+      Print.punctuation ppf " =";
+      Format.pp_close_box ppf ();
```

Comment:
Why don't you use the `Format` specification syntax with `@[` ?

Reply:
See my answer [below](https://github.com/CatalaLang/catala/pull/391#discussion_r1106123928)